platforms/jvm/language-java/src/integTest/groovy/org/gradle/api/tasks/JavaExecWithLongCommandLineIntegrationTest.groovy

```diff
@@ -66,7 +68,9 @@ class JavaExecWithLongCommandLineIntegrationTest extends AbstractIntegrationSpec
                 def execOps = services.get(ExecOperations)
                 doLast {
                     execOps.javaexec {
-                        executable = runExecutable
+                        if (runExecutable.isPresent()) {
```

Comment:
💭 This reverts behaviour to the same as is in `master`. If we set executable to empty provider, then execution will fail since executable is not set